### PR TITLE
Conversion to v6

### PR DESCRIPTION
**Pull request by petersgiles**
https://github.com/Particular/NServiceBus.Host/pull/40

blocked by https://github.com/Particular/NServiceBus/issues/3238

### CODE REVIEW

**Comment by petersgiles** [comment]
@andreasohlund 

It seems that the new Core doesn't default to MsmqTransport, is that by design or a problem?

```
        [Test]
        public void Should_default_to_msmq_if_no_other_transport_is_configured()
        {
            var builder = new BusConfiguration();
            builder.EndpointName("myTests");

            Assert.True(builder.GetSettings().Get<TransportDefinition>() is MsmqTransport);            
        }

```

**Comment by petersgiles** [comment]
ping @andreasohlund

**Comment by petersgiles** [comment]
@SimonCropp @andreasohlund @danielmarbach ready for merge, please review.

File: src/NServiceBus.Hosting.Profiles/IConfigureLoggingForProfile.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@
     /// </summary>
     /// <remarks>
     /// If you want logging configured regardless of profiles, do not use this interface,
-    /// instead configure logging before you call <see cref="Bus.Create"/> if you self hosting or configure logging the constructor of the class that implements <see cref="IConfigureThisEndpoint"/>.
+    /// instead configure logging before you call <see cref="Endpoint.Create"/> if you self hosting or configure logging the constructor of the class that implements <see cref="IConfigureThisEndpoint"/>.
```
Comment (SimonCropp) [inline src/NServiceBus.Hosting.Profiles/IConfigureLoggingForProfile.cs]: doco on a obsolete thing is redundant

**Comment by SimonCropp** [comment]
have u smoke tested it on any of the doco samples?

File: src/NServiceBus.Hosting.Windows/GenericHost.cs [hunk 2]
```diff
@@ -46,19 +46,18 @@ public GenericHost(IConfigureThisEndpoint specifier, string[] args, List<Type> d
         /// <summary>
         ///     Creates and starts the bus as per the configuration
         /// </summary>
-        public void Start()
+        public async void Start()
         {
             try
             {
-                bus = PerformConfiguration().Start();
+                var startableEndpoint = await PerformConfiguration();
```
Comment (SimonCropp) [inline src/NServiceBus.Hosting.Windows/GenericHost.cs]: add `ConfigureAwait(false)`

File: src/NServiceBus.Hosting.Windows/GenericHost.cs [hunk 3]
```diff
@@ -132,19 +129,20 @@ internal static bool TryGetSlaFromEndpointConfigType(Type endpointConfigurationT
...
             
-            Environment.FailFast($"The following critical error was encountered by NServiceBus:\n{errorMessage}\nNServiceBus is shutting down.", exception);
+            Environment.FailFast($"The following critical error was encountered by NServiceBus:\n{error}\nNServiceBus is shutting down.", exception);
+            await Task.FromResult(0);
         }
 
         ProfileManager profileManager;
         IConfigureThisEndpoint specifier;
-        IBus bus;
+        Task<IEndpointInstance> bus;
```
Comment (SimonCropp) [inline src/NServiceBus.Hosting.Windows/GenericHost.cs]: why is the task stored and not the instance?

File: src/NServiceBus.Hosting.Profiles/IConfigureLoggingForProfile.cs (hunk 1 above)
Comment (petersgiles) [inline src/NServiceBus.Hosting.Profiles/IConfigureLoggingForProfile.cs]: cool will remove

File: src/NServiceBus.Hosting.Windows/GenericHost.cs (hunk 3 above)
Comment (petersgiles) [inline src/NServiceBus.Hosting.Windows/GenericHost.cs]: fair point, will fix

**Pull request merged**